pkg/controller/operators/olm/requirements.go

```diff
@@ -72,118 +71,9 @@ func (a *Operator) minKubeVersionStatus(name string, minKubeVersion string) (met
 
 func (a *Operator) requirementStatus(strategyDetailsDeployment *v1alpha1.StrategyDetailsDeployment, crdDescs []v1alpha1.CRDDescription,
 	ownedAPIServiceDescs []v1alpha1.APIServiceDescription, requiredAPIServiceDescs []v1alpha1.APIServiceDescription,
-	requiredNativeAPIs []metav1.GroupVersionKind) (met bool, statuses []v1alpha1.RequirementStatus) {
+	requiredNativeAPIs []metav1.GroupVersionKind) (met bool, statuses []v1alpha1.RequirementStatus, err error) {
 	met = true
 
-	// Check for CRDs
-	for _, r := range crdDescs {
-		status := v1alpha1.RequirementStatus{
-			Group:   "apiextensions.k8s.io",
-			Version: "v1beta1",
-			Kind:    "CustomResourceDefinition",
-			Name:    r.Name,
-		}
-
-		// check if CRD exists - this verifies group, version, and kind, so no need for GVK check via discovery
-		crd, err := a.lister.APIExtensionsV1beta1().CustomResourceDefinitionLister().Get(r.Name)
-		if err != nil {
-			status.Status = v1alpha1.RequirementStatusReasonNotPresent
-			status.Message = "CRD is not present"
-			a.logger.Debugf("Setting 'met' to false, %v with status %v, with err: %v", r.Name, status, err)
-			met = false
-			statuses = append(statuses, status)
-			continue
-		}
-
-		if crd.Spec.Version != r.Version {
-			served := false
-			for _, version := range crd.Spec.Versions {
-				if version.Name == r.Version {
-					if version.Served {
```

Comment:
Why was this removed?

Reply:
This tries to see if the crdDesc has the version that the CRD's api has. If we are already checking the gvk from the CRDDesc, then this is not necessary anymore, right?

Reply:
I do not believe that this is true. It is possible that a single CRD contains multiple versions that are `served`. For example, imaging that you are upgrading your CRD from v1 to v2. Only one of these versions of the CRD are stored in etcd/object store. However, there are cases where I should still be able to create/update/delete v1 of the CRD.

Check out bullet point 1 in [this doc](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#overview)

Reply:
Yes, but these are CRD Descriptions which are singletons. Under the same CRD, different versions have different descriptions. At the very least the descriptions would have different versions.